File: 2022/day_04.py
import parse

import aoc_lube

RANGES = tuple(parse.findall("{:d}-{:d},{:d}-{:d}", aoc_lube.fetch(year=2022, day=4)))

def part_one():
    return sum(a <= c and d <= b or c <= a and b <= d for a, b, c, d, in RANGES)

def part_two():
    return sum(a <= d and c <= b for a, b, c, d in RANGES)

aoc_lube.submit(year=2022, day=4, part=1, solution=part_one)
aoc_lube.submit(year=2022, day=4, part=2, solution=part_two)

File: 2022/day_04.nim
include aoc

let RANGES = block:
  var a, b, c, d: int
  collect(
    for line in fetch(2022, 4).splitLines:
      discard line.scanf("$i-$i,$i-$i", a, b, c, d)
      (a, b, c, d)
  )

part 1: sum collect(
  for (a, b, c, d) in RANGES:
    int a <= c and d <= b or c <= a and b <= d
)

part 2: sum collect(
  for (a, b, c, d) in RANGES:
    int a <= d and c <= b
)